crates/apub/src/objects/post.rs

```diff
@@ -76,10 +76,12 @@ impl ApubObject for ApubPost {
   }
 
   async fn delete(self, context: &LemmyContext) -> Result<(), LemmyError> {
-    blocking(context.pool(), move |conn| {
-      Post::update_deleted(conn, self.id, true)
-    })
-    .await??;
+    if !self.deleted {
+      blocking(context.pool(), move |conn| {
+        Post::update_deleted(conn, self.id, true)
+      })
+      .await??;
+    }
```

Comment:
Why dont we just do this check inside of Post::update_deleted?

Reply:
I'm kinda against adding external logic / checks to DB commands / functions, it only adds complications when you need to not do those checks elsewhere.

Reply:
But doing the same check in apub and api code is also not great. Maybe we need some intermediate layer, over the db code but below api/apub (i think i suggested that before).

Reply:
I remember talking about that a while back, but if we do it, it could be done another time.

Reply:
Yes for sure.